### PR TITLE
Add Self signup configurations page.

### PR DESCRIPTION
**Pull request by vihanga-liyanage**
image

Refers https://github.com/wso2/identity-apps/issues/581

### CODE REVIEW

File: apps/admin-portal/src/components/serverConfigurations/account-recovery.tsx [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+		</Modal>
+	);
+
+	const showUserSelfRegistrationView = editingForm[ACCOUNT_RECOVERY_FORM_IDENTIFIER] ? (
+		<EditSection>
+			<Forms
+				onSubmit={ (value) => {
+					// Todo complete method
+					// setCurrentPassword(value.get("currentPassword").toString());
+					console.log(value);
```
Comment (thivi) [inline apps/admin-portal/src/components/serverConfigurations/account-recovery.tsx]:
```suggestion
```

File: apps/admin-portal/src/models/server-configurations.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ *  Captures the basic details in the applications.
+ */
+export interface SelfSignUpConfigurationsInterface {
+
```
Comment (thivi) [inline apps/admin-portal/src/models/server-configurations.ts]: Remove this line

File: apps/admin-portal/src/configs/app.ts [hunk 3]
```diff
@@ -71,5 +76,6 @@ export const ServiceResourcesEndpoint: ServiceResourcesType = {
     user: `${GlobalConfig.serverHost}/api/identity/user/v1.0/me`,
     userStores: `${GlobalConfig.serverHost}/api/server/v1/userstores`,
     users: `${GlobalConfig.serverHost}/scim2/Users`,
-    wellKnown: `${GlobalConfig.serverHost}/oauth2/oidcdiscovery/.well-known/openid-configuration`
+    wellKnown: `${GlobalConfig.serverHost}/oauth2/oidcdiscovery/.well-known/openid-configuration`,
+    selfSignUp: `${GlobalConfig.serverHost}/api/server/v1/identity-governance/${IDENTITY_GOVERNANCE_ACCOUNT_MANAGEMENT_POLICIES_ID}/connectors/${SELF_SIGN_UP_CONNECTOR_ID}`
```
Comment (thivi) [inline apps/admin-portal/src/configs/app.ts]: I think this line exceeds 120 characters.

File: apps/admin-portal/src/components/serverConfigurations/account-recovery.tsx [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+				/>
+				<h4>Password Recovery</h4>
+				<Field
+					name="PasswordRecoveryOptions"
+					required={ false }
+					requiredErrorMessage=""
+					type="checkbox"
+					children={ [
+						{
+							label: t("views:components.serverConfigs.accountRecovery.passwordRecovery.form.enableNotificationBasedRecovery.label"),
```
Comment (thivi) [inline apps/admin-portal/src/components/serverConfigurations/account-recovery.tsx]: This line exceeds 120 characters. Please check the other lines as well.

File: apps/admin-portal/src/configs/app.ts (hunk 3 above)
Comment (vihanga-liyanage) [reply apps/admin-portal/src/configs/app.ts]: Tried breaking this to two but then it gives an error. How to break this?

Comment (thivi) [reply apps/admin-portal/src/configs/app.ts]: You can use `\` to ensure line continuation. 
```suggestion
    selfSignUp: `${GlobalConfig.serverHost}/api/server/v1/identity-governance/\
    ${IDENTITY_GOVERNANCE_ACCOUNT_MANAGEMENT_POLICIES_ID}/connectors/${SELF_SIGN_UP_CONNECTOR_ID}`
```

File: apps/admin-portal/src/pages/server-configurations.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { FunctionComponent, ReactElement } from "react";
+import { useDispatch } from "react-redux";
+import { PageLayout } from "../layouts";
+import { Grid } from "semantic-ui-react";
+import { AlertInterface } from "../../../user-portal/src/models";
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/pages/server-configurations.tsx]:
```suggestion
import { AlertInterface } from "@wso2is/core/models";
```

File: apps/admin-portal/src/pages/server-configurations.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import { addAlert } from "../../../user-portal/src/store/actions";
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/pages/server-configurations.tsx]:
```suggestion
import { addAlert } from "@wso2is/core/store";
```

File: apps/admin-portal/src/api/user-self-registration.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * Initialize an axios Http client.
+ * @type { AxiosHttpClientInstance }
+ */
+const httpClient = AxiosHttpClient.getInstance();
+
+/**
+ * Retrieve self sign up configurations.
+ *
+ * @returns {Promise<BasicProfileInterface>} a promise containing the configurations.
```
Comment (brionmario) [inline apps/admin-portal/src/api/user-self-registration.ts]:
```suggestion
 * @returns {Promise<any>} a promise containing the configurations.
```

File: apps/admin-portal/src/api/user-self-registration.ts [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+            return Promise.reject(error);
+        });
+};
+
+/**
+ * Update self sign up configurations.
+ *
+ * @param data request payload
+ *
+ * @returns {Promise<BasicProfileInterface>} a promise containing the response.
```
Comment (brionmario) [inline apps/admin-portal/src/api/user-self-registration.ts]:
```suggestion
 * @returns {Promise<any>} a promise containing the response.
```

File: apps/admin-portal/src/components/serverConfigurations/account-recovery.tsx [hunk 9]
```diff
@@ -0,0 +1,408 @@
...
+
+	/**
+	 * Calls the API and updates the user password.
+	 */
+	const saveSelfRegistrationConfigs = () => {
+		// Todo Implement method.
+	};
+
+	const confirmationModal = (
+		<Modal size="mini" open={ showConfirmationModal } onClose={ handleConfirmationModalClose } dimmer="blurring">
```
Comment (brionmario) [inline apps/admin-portal/src/components/serverConfigurations/account-recovery.tsx]: There seems to be a max length issue here. Fix other places as well.

File: apps/admin-portal/src/configs/app.ts (hunk 3 above)
Comment (brionmario) [reply apps/admin-portal/src/configs/app.ts]: Or just break like bellow.

```suggestion
    selfSignUp: `${GlobalConfig.serverHost}/api/server/v1/identity-governance/${
    IDENTITY_GOVERNANCE_ACCOUNT_MANAGEMENT_POLICIES_ID}/connectors/${SELF_SIGN_UP_CONNECTOR_ID}`
```

File: apps/admin-portal/src/pages/server-configurations.tsx [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+import {
+	AccountRecovery,
+	UserSelfRegistration
+} from "../components/serverConfigurations";
```
Comment (brionmario) [inline apps/admin-portal/src/pages/server-configurations.tsx]: Shall we export these components from the index of the components folder?

File: apps/admin-portal/src/components/serverConfigurations/account-recovery.tsx (hunk 1 above)
Comment (vihanga-liyanage) [reply apps/admin-portal/src/components/serverConfigurations/account-recovery.tsx]: Fixed.

File: apps/admin-portal/src/configs/app.ts (hunk 3 above)
Comment (vihanga-liyanage) [reply apps/admin-portal/src/configs/app.ts]: Fixed.

**Pull request merged**